### PR TITLE
Replace unmaintained Bintray release upload script with `com.novoda.bintray-release`

### PR DESCRIPTION
**Pull request by KevinLeighCrain**
Suggested by @blundell

### CODE REVIEW

**Comment by KevinLeighCrain** [comment]
@blundell Do you know what the reasons would be for me receiving message `Could not find publication: maven.` when running `bintrayUpload` gradle task? Note I set `dryRun` to false to prevent publishing, would that possibly be the reason?

**Comment by blundell** [comment]
looks like the last merge () has got some issues, use version `0.3.4` which works perfectly

**Comment by blundell** [comment]
also note `dryRun=true` will prevent publishing :-) not sure if you had a typo above ^

**Comment by KevinLeighCrain** [comment]
Thanks I have bumped down the dependency to `0.3.4`, I presume I need to have a sonatype account for  the publishing of this library to work against the script?

**Comment by blundell** [comment]
to release locally you don't need any type of account - you just use the param `-PdryRun=true` (you might have to supply user and key but they can just be like (`-PbintrayUser=zzz -PbintrayKey=123`)

to release to jcenter you don't need a sonatype account you need a bintray account (which you can just sign in with GitHub). After signing up go here https://bintray.com/profile/edit and go to `API key` and use that  with `-PdryRun=false` and `-PbintrayKey=yourApiKey` (and don't forget `-PbintrayUser=` to release to jcenter.

**Comment by KevinLeighCrain** [comment]
I have a Bintray account and have successfully uploaded a package manually to my hosted Maven repo and then published the package to jCenter, Why I asked question above, is because the script outputs message `Could not find publication: maven.`, do you possibly from your experience know the reason? I do see the message resides somewhere in the nodova gradle plugin script when I search the repository

**Comment by blundell** [comment]
I believe @xrigau knows the answer

**Comment by xrigau** [comment]
Hey @KevinLeighCrain ! I've just checked out this branch and did 
`gradle clean build bintrayUpload -PbintrayUser=xrigau -PbintrayKey=secret_key` 
and I was able to publish it https://bintray.com/xrigau/maven/chartify/0.9/view

which command are you using to publish?

btw I'll delete that package :wink:

**Comment by KevinLeighCrain** [comment]
@xrigau I am using a gradle wrapper with the following command `./gradlew clean build bintrayUpload -PbintrayUser=kevinleighcrain -PbintrayKey=secret_key`

Thanks ;)

**Comment by xrigau** [comment]
@KevinLeighCrain the problem might be that we try to use the default repository name (`repoName `) which is `maven` but doesn't seem like you have any repo called `maven` in your bintray account

So I guess you can either create one or change that property in the configuration

https://github.com/novoda/bintray-release/wiki/Configuration-of-the-publish-closure look for `repoName`

Hope that helps

**Comment by KevinLeighCrain** [comment]
@xrigau It worked! I was missing I need to add the `repoName` Thanks to you and @blundell for this learning curve, will definitely boost my releases in the future, much appreciated :+1:

File: library/build.gradle [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 
-ext {
-    PUBLISH_GROUP_ID = 'com.kevinleighcrain'
-    PUBLISH_ARTIFACT_ID =  'chartify'
-    PUBLISH_VERSION = '0.9'
+Properties properties = new Properties()
+properties.load(project.rootProject.file('local.properties').newDataInputStream())
+
+publish {
+    repoName = properties.getProperty("bintray.userName")
```
Comment (KevinLeighCrain) [inline library/build.gradle]: Typo :|

**Comment by blundell** [comment]
:100: nice spot @xrigau

**Comment by xrigau** [comment]
Glad it worked! :+1:

**Pull request merged**